rust/core-lib/src/editor.rs

```diff
@@ -685,6 +686,48 @@ impl Editor {
         }
     }
 
+    // capitalization behaviour is similar to behaviour in XCode
+    fn capitalize_text(&mut self, view: &mut View) {
+        let mut builder = delta::Builder::new(self.text.len());
+        let mut final_selection = Selection::new();
+
+        for &region in view.sel_regions() {
+            final_selection.add_region(SelRegion::new(region.max(), region.max()));
+            let mut word_cursor = WordCursor::new(&self.text, region.min());
+
+            loop {
+                // capitalize each word in the current selection
+                let (start, end) = word_cursor.select_word();
+
+                if start < end {
+                    let interval = Interval::new_closed_open(start, end);
+                    let word = self.text.slice_to_string(start..end);
+
+                    // first letter is uppercase, remaining letters are lowercase
+                    let (first_char, rest) = word.split_at(1);
+                    let mut capitalized_text = String::from("");
```

Comment:
note/non-blocking: you can get an empty string with `String::new()`; and if we know approximately our length (ignoring, say, ß) we might save ourselves an allocation here with `String::with_capacity(word.len())`.

Reply:
I think [a, b].concat() is the preferred idiom here.